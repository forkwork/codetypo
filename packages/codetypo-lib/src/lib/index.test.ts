import { describe, expect, test } from 'vitest';

import * as codetypo from './index.js';

describe('Validate the codetypo API', () => {
    test('Tests the default configuration', async () => {
        const ext = '.json';
        const languageIds = codetypo.getLanguagesForExt(ext);
        const settings = await codetypo.getDefaultBundledSettingsAsync();
        // codetypo:ignore jansons
        const text = '{ "name": "Jansons"}';
        const fileSettings = codetypo.combineTextAndLanguageSettings(settings, text, languageIds);
        return codetypo.validateText(text, fileSettings).then((results) => {
            expect(Object.keys(results)).not.toHaveLength(0);
            expect(results.map((to) => to.text)).toEqual(expect.arrayContaining(['Jansons']));
            return;
        });
    });

    test('Verify API exports', () => {
        expect(codetypo).toMatchSnapshot();
    });
});
