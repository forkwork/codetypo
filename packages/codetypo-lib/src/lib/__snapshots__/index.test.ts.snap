// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Validate the codetypo API > Verify API exports 1`] = `
{
  "CheckTextInfo": undefined,
  "CompoundWordsMethod": {
    "0": "NONE",
    "1": "SEPARATE_WORDS",
    "2": "JOIN_WORDS",
    "JOIN_WORDS": 2,
    "NONE": 0,
    "SEPARATE_WORDS": 1,
  },
  "ConfigFields": {
    "allowCompoundWords": "allowCompoundWords",
    "cache": "cache",
    "caseSensitive": "caseSensitive",
    "description": "description",
    "dictionaries": "dictionaries",
    "dictionaryDefinitions": "dictionaryDefinitions",
    "enableFiletypes": "enableFiletypes",
    "enableGlobDot": "enableGlobDot",
    "enabled": "enabled",
    "enabledFileTypes": "enabledFileTypes",
    "enabledLanguageIds": "enabledLanguageIds",
    "failFast": "failFast",
    "features": "features",
    "files": "files",
    "flagWords": "flagWords",
    "gitignoreRoot": "gitignoreRoot",
    "globRoot": "globRoot",
    "ignorePaths": "ignorePaths",
    "ignoreRandomStrings": "ignoreRandomStrings",
    "ignoreRegExpList": "ignoreRegExpList",
    "ignoreWords": "ignoreWords",
    "import": "import",
    "includeRegExpList": "includeRegExpList",
    "language": "language",
    "languageId": "languageId",
    "languageSettings": "languageSettings",
    "loadDefaultConfiguration": "loadDefaultConfiguration",
    "maxDuplicateProblems": "maxDuplicateProblems",
    "maxNumberOfProblems": "maxNumberOfProblems",
    "minRandomLength": "minRandomLength",
    "minWordLength": "minWordLength",
    "name": "name",
    "noConfigSearch": "noConfigSearch",
    "noSuggestDictionaries": "noSuggestDictionaries",
    "numSuggestions": "numSuggestions",
    "overrides": "overrides",
    "parser": "parser",
    "patterns": "patterns",
    "pnpFiles": "pnpFiles",
    "readonly": "readonly",
    "reporters": "reporters",
    "showStatus": "showStatus",
    "spellCheckDelayMs": "spellCheckDelayMs",
    "suggestWords": "suggestWords",
    "suggestionNumChanges": "suggestionNumChanges",
    "suggestionsTimeout": "suggestionsTimeout",
    "useGitignore": "useGitignore",
    "usePnP": "usePnP",
    "userWords": "userWords",
    "validateDirectives": "validateDirectives",
    "words": "words",
  },
  "DetermineFinalDocumentSettingsResult": undefined,
  "DocumentValidator": [Function],
  "DocumentValidatorOptions": undefined,
  "ENV_CODETYPO_GLOB_ROOT": "CODETYPO_GLOB_ROOT",
  "ExcludeFilesGlobMap": undefined,
  "ExclusionFunction": undefined,
  "ExclusionHelper": {
    "extractGlobsFromExcludeFilesGlobMap": [Function],
    "generateExclusionFunctionForFiles": [Function],
    "generateExclusionFunctionForUri": [Function],
  },
  "FSCapabilityFlags": {
    "0": "None",
    "1": "Stat",
    "16": "WriteDir",
    "2": "Read",
    "24": "ReadWriteDir",
    "4": "Write",
    "6": "ReadWrite",
    "8": "ReadDir",
    "None": 0,
    "Read": 2,
    "ReadDir": 8,
    "ReadWrite": 6,
    "ReadWriteDir": 24,
    "Stat": 1,
    "Write": 4,
    "WriteDir": 16,
  },
  "FeatureFlag": undefined,
  "FeatureFlags": [Function],
  "ImportError": [Function],
  "IncludeExcludeFlag": {
    "EXCLUDE": "E",
    "INCLUDE": "I",
  },
  "IncludeExcludeOptions": undefined,
  "IssueType": {
    "0": "spelling",
    "1": "directive",
    "directive": 1,
    "spelling": 0,
  },
  "Link": {
    "addPathsToGlobalImports": [Function],
    "listGlobalImports": [Function],
    "removePathsFromGlobalImports": [Function],
  },
  "Logger": undefined,
  "MessageTypes": {
    "Debug": "Debug",
    "Info": "Info",
    "Warning": "Warning",
  },
  "SpellCheckFileOptions": undefined,
  "SpellCheckFilePerf": undefined,
  "SpellCheckFileResult": undefined,
  "SpellingDictionary": undefined,
  "SpellingDictionaryCollection": undefined,
  "SpellingDictionaryLoadError": [Function],
  "SuggestOptions": undefined,
  "SuggestionCollector": undefined,
  "SuggestionError": [Function],
  "SuggestionResult": undefined,
  "Text": {
    "calculateTextDocumentOffsets": [Function],
    "camelToSnake": [Function],
    "cleanText": [Function],
    "cleanTextOffset": [Function],
    "extractLinesOfText": [Function],
    "extractPossibleWordsFromTextOffset": [Function],
    "extractText": [Function],
    "extractWordsFromCode": [Function],
    "extractWordsFromCodeTextOffset": [Function],
    "extractWordsFromText": [Function],
    "extractWordsFromTextOffset": [Function],
    "isFirstCharacterLower": [Function],
    "isFirstCharacterUpper": [Function],
    "isLowerCase": [Function],
    "isUpperCase": [Function],
    "lcFirst": [Function],
    "match": [Function],
    "matchCase": [Function],
    "matchStringToTextOffset": [Function],
    "matchToTextOffset": [Function],
    "removeAccents": [Function],
    "snakeToCamel": [Function],
    "splitCamelCaseWord": [Function],
    "splitCamelCaseWordWithOffset": [Function],
    "stringToRegExp": [Function],
    "textOffset": [Function],
    "ucFirst": [Function],
  },
  "TextInfoItem": undefined,
  "UnknownFeatureFlagError": [Function],
  "ValidationIssue": undefined,
  "asyncIterableToArray": [Function],
  "calcOverrideSettings": [Function],
  "checkFilenameMatchesGlob": [Function],
  "checkText": [Function],
  "checkTextDocument": [Function],
  "clearCachedFiles": [Function],
  "clearCaches": [Function],
  "combineTextAndLanguageSettings": [Function],
  "constructSettingsForText": [Function],
  "createConfigLoader": [Function],
  "createPerfTimer": [Function],
  "createSpellingDictionary": [Function],
  "createSpellingDictionaryCollection": [Function],
  "createTextDocument": [Function],
  "currentSettingsFileVersion": "0.2",
  "defaultCodeTypoSettings": {
    "ignoreRandomStrings": true,
    "minRandomLength": 40,
  },
  "defaultConfigFilenames": [
    "package.json",
    ".codetypo.json",
    "codetypo.json",
    ".codeTypo.json",
    "codeTypo.json",
    ".codetypo.jsonc",
    "codetypo.jsonc",
    ".vscode/codetypo.json",
    ".vscode/codeTypo.json",
    ".vscode/.codetypo.json",
    ".codetypo.config.json",
    ".codetypo.config.jsonc",
    ".codetypo.config.yaml",
    ".codetypo.config.yml",
    "codetypo.config.json",
    "codetypo.config.jsonc",
    "codetypo.config.yaml",
    "codetypo.config.yml",
    "codetypo.config.mjs",
    "codetypo.config.cjs",
    "codetypo.config.js",
    ".codetypo.config.mjs",
    ".codetypo.config.cjs",
    ".codetypo.config.js",
    ".codetypo.yaml",
    ".codetypo.yml",
    "codetypo.yaml",
    "codetypo.yml",
    ".config/.codetypo.json",
    ".config/codetypo.json",
    ".config/.codeTypo.json",
    ".config/codeTypo.json",
    ".config/.codetypo.jsonc",
    ".config/codetypo.jsonc",
    ".config/codetypo.config.json",
    ".config/codetypo.config.jsonc",
    ".config/codetypo.config.yaml",
    ".config/codetypo.config.yml",
    ".config/codetypo.config.mjs",
    ".config/codetypo.config.cjs",
    ".config/codetypo.config.js",
    ".config/.codetypo.config.json",
    ".config/.codetypo.config.jsonc",
    ".config/.codetypo.config.yaml",
    ".config/.codetypo.config.yml",
    ".config/.codetypo.config.mjs",
    ".config/.codetypo.config.cjs",
    ".config/.codetypo.config.js",
    ".config/codetypo.yaml",
    ".config/codetypo.yml",
  ],
  "defaultFileName": "codetypo.json",
  "defaultSettingsFilename": "codetypo.json",
  "defineConfig": [Function],
  "determineFinalDocumentSettings": [Function],
  "extractDependencies": [Function],
  "extractImportErrors": [Function],
  "fileToDocument": [Function],
  "fileToTextDocument": [Function],
  "finalizeSettings": [Function],
  "getCachedFileSize": [Function],
  "getDefaultBundledSettingsAsync": [Function],
  "getDefaultConfigLoader": [Function],
  "getDefaultSettings": [Function],
  "getDictionary": [Function],
  "getGlobalSettings": [Function],
  "getGlobalSettingsAsync": [Function],
  "getLanguageIdsForBaseFilename": [Function],
  "getLanguagesForExt": [Function],
  "getLogger": [Function],
  "getSources": [Function],
  "getSystemFeatureFlags": [Function],
  "getVirtualFS": [Function],
  "isBinaryFile": [Function],
  "isSpellingDictionaryLoadError": [Function],
  "loadConfig": [Function],
  "loadPnP": [Function],
  "mergeInDocSettings": [Function],
  "mergeSettings": [Function],
  "readConfigFile": [Function],
  "readFile": [Function],
  "readFileSync": [Function],
  "readRawSettings": [Function],
  "readSettings": [Function],
  "readSettingsFiles": [Function],
  "refreshDictionaryCache": [Function],
  "resolveConfigFileImports": [Function],
  "resolveFile": [Function],
  "searchForConfig": [Function],
  "sectionCodeTypo": "codeTypo",
  "setLogger": [Function],
  "shouldCheckDocument": [Function],
  "spellCheckDocument": [Function],
  "spellCheckFile": [Function],
  "suggestionsForWord": [Function],
  "suggestionsForWords": [Function],
  "traceWords": [Function],
  "traceWordsAsync": [Function],
  "updateTextDocument": [Function],
  "validateText": [Function],
  "writeToFile": [Function],
  "writeToFileIterable": [Function],
  "writeToFileIterableP": [Function],
}
`;
